### PR TITLE
Reorder extended startup to yield jUnit with TEST_ONLY

### PR DESCRIPTION
**Pull request by stevekuznetsov**
The current ordering of the parsing of `$JUNIT_REPORT` and `$TEST_ONLY`
means that no jUnit is generated or placed when running with both
options. This patch re-orders the parsing so that jobs running extended
tests against pre-existing clusters get jUnit output as well.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

[test]
@smarterclayton PTAL

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Testing also in https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_extended_conformance_install/1/

**Comment by stevekuznetsov** [comment]
https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_extended_conformance_install/2/

**Comment by stevekuznetsov** [comment]
@smarterclayton could you please TAL

**Comment by stevekuznetsov** [comment]
[merge]

**Comment by stevekuznetsov** [comment]
[test_install]

File: test/extended/setup.sh [hunk 1]
```diff
@@ -33,19 +33,19 @@ function os::test::extended::setup () {
...
-
 	os::util::environment::setup_time_vars
 	os::util::environment::use_sudo
 	os::util::environment::setup_all_server_vars "test-extended/core"
 
 	os::util::ensure::iptables_privileges_exist
 
+	# Allow setting $JUNIT_REPORT to toggle output behavior
+	if [[ -n "${JUNIT_REPORT:-}" ]]; then
+		export JUNIT_REPORT_OUTPUT="${LOG_DIR}/raw_test_output.log"
```
Comment (stevekuznetsov) [inline test/extended/setup.sh]: @stevekuznetsov need to touch this to make sure it exists? empty if no `os::cmd` stuff runs

**Comment by stevekuznetsov** [comment]
https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin/13/

**Comment by stevekuznetsov** [comment]
Looks like we get jUnit correctly in both GCE and the new install test ... [merge]

**Comment by smarterclayton** [comment]
This broke TEST_ONLY... because i don't need root to run the extended tests, or to gather output from them.

I'll open a PR tomorrow that splits tmpdir_vars from server (the two have nothing to do with each other).  It would probably be better if we stopped using /tmp at all, and instead it was just a relative directory inside our repo that depended on the initial script you call.  Not going to make that big a change though now.

**Comment by stevekuznetsov** [comment]
> I'll open a PR tomorrow that splits tmpdir_vars from server

Yeah, we should have in `hack/lib/init.sh` some `$TMPDIR` setup that can hold server stuff but doesn't need to. Agreed.

> instead it was just a relative directory inside our repo 

We were kind of doing that with `_output` for generate and verify steps, but we need to be careful there because people may be doing `make clean` in places that would nuke their dirs in-flight etc.

**Pull request merged**